code/modules/jobs/job_types/_job.dm

```diff
@@ -132,10 +132,29 @@
 					continue
 
 				if(G.slot)
-					if(H.equip_to_slot_or_del(G.spawn_item(H), G.slot))
-						to_chat(M, "<span class='notice'>Equipping you with [G.display_name]!</span>")
+					if(G.slot == ITEM_SLOT_BACK)
+						var/obj/item/storage/backpack/oldbag = H.get_item_by_slot(ITEM_SLOT_BACK)
+						var/metadata = M.client.prefs.active_character.equipped_gear[G.id]
+						var/obj/item/storage/backpack/newbag = G.spawn_item(null, metadata)
+						//This was much cleaner than trying to safely transfer contents in the correct orientation
```

Comment:
I hate this, it's awful, can you not just use the same logic that dragging one bag onto another does?

Reply:
if you can point me to the proc, sure. 
I really like being able to make XL versions of the donor shop bags though.

Reply:
it should be storage component's response to the drag release signal

Reply:
Is there some particular reason you hate simply overwriting the appearance of the bag when it's functionally better in the end due to allowing more customization than before?

Reply:
Found the proc, know how to do this now. I'm just left not wanting to because I like what I have here more.

Will follow through with changing it if you want to apply pressure to, but I really like overwriting the appearance more than doing an item transfer.

Reply:
Rewriting the appearance of a bag is just asking for trouble imo

Reply:
I mean, that's all the donor bags are really meant to be anyway - alternative bag skins?

Except for the distinction between duffel and non-duffel (which is accounted for here) the only thing that differentiates bags is their name, descriptions and icons.

Reply:
UUugh, it just fills me with pain to have something's typepath not be what it actually is, it seems illegal

I suppose it works though since we are just reskinning it... the PDA colorizer works the same way. Alright, fine, it's fine I guess.